qemu/tests/hotplug_mem_share_discard_data.py

```diff
@@ -0,0 +1,88 @@
+from avocado.utils import process
+
+from virttest import error_context
+from virttest import env_process
+
+from virttest.utils_numeric import normalize_data_size
+from virttest.utils_test.qemu import MemoryHotplugTest
+from virttest.staging import utils_memory
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Memory share and discard-data hotplug test
+    1) Setup hugepages
+    2) Mount /mnt/kvm_hugepage
+    3) Boot guest with one numa node
+    4) Hotplug 1G memory
+    5) Check memory
+    6) Unplug memory
+    7) Check memory
+    8) Check results for discard-data value
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    timeout = float(params.get("login_timeout", 240))
+
+    mem_dev = params.get('mem_devs')
+    size_mem = int(normalize_data_size(params['size_mem_%s' % mem_dev], 'K'))
+    total_hg_size = size_mem
+
+    target_mem = params.get("target_mems")
+    if params.get('backend_mem_%s' % target_mem) == 'memory-backend-file':
+        size_target_mem = int(normalize_data_size(params['size_mem_%s' % target_mem], 'K'))
+        total_hg_size += size_target_mem
+
+    hp_size = utils_memory.read_from_meminfo("Hugepagesize")
+    params['target_hugepages'] = int(total_hg_size // hp_size)
+    params['setup_hugepages'] = "yes"
+    params['not_preprocess'] = "no"
+
+    env_process.preprocess(test, params, env)
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+
+    hotplug_test = MemoryHotplugTest(test, params, env)
+    hotplug_test.hotplug_memory(vm, target_mem)
+    hotplug_test.check_memory(vm)
+    hotplug_test.unplug_memory(vm, target_mem)
+    hotplug_test.check_memory(vm)
+
+    session.close()
+    error_context.context("Shutdown guest...")
+    vm.destroy()
+
+    hp_total = int(utils_memory.read_from_meminfo("HugePages_Total"))
+    hp_free = int(utils_memory.read_from_meminfo('HugePages_Free'))
+
+    error_context.context("hp_total: %s" % str(hp_total), test.log.debug)
+    error_context.context("hp_free: %s" % str(hp_free), test.log.debug)
+
+    if params.get("backend_mem_plug1") == "memory-backend-file":
+        if not params.get_boolean("discard-data_plug1", True):
```

Comment:
L67-L68
discard-data should only support backend_file mode, so we can reduce one judgment

`if params.get("discard-data_plug1") == "no"`

Reply:
The other part is ack for me

```
 (1/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.hotplug_mem_share_discard_data.backend_ram.arm64-pci: STARTED
 (1/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.hotplug_mem_share_discard_data.backend_ram.arm64-pci: PASS (122.20 s)
 (2/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.hotplug_mem_share_discard_data.backend_file.discard_data_on.arm64-pci: STARTED
 (2/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.hotplug_mem_share_discard_data.backend_file.discard_data_on.arm64-pci: PASS (129.91 s)
 (3/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.hotplug_mem_share_discard_data.backend_file.discard_data_off.arm64-pci: STARTED
 (3/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.hotplug_mem_share_discard_data.backend_file.discard_data_off.arm64-pci: PASS (119.48 s)
```

Reply:
Thanks for the suggestions!

Reply:
Changed